scan_struct.go

```diff
@@ -1,17 +1,61 @@
 package pgxhelpers
 
 import (
+	"context"
 	"errors"
 	"fmt"
 	"reflect"
 	"strings"
+	"unsafe"
 
+	"github.com/jackc/pgconn"
 	"github.com/jackc/pgproto3/v2"
 	"github.com/jackc/pgx/v4"
 	"github.com/jmoiron/sqlx/reflectx"
 )
 
 var mapper = reflectx.NewMapperFunc("db", strings.ToLower)
+var rowsType reflect.Type
+
+func init() {
+	// XXX: hardcore unsafe part 1, just to make it work, not for production
+	// see part 2 below to see why I need all this
+
+	// all this unsafe black magic is required to get the instance of private *pgx.connRows for further usage
+	// to do this - need to initialise fake connection with the minimal set of fields set just to make it return
+	// instance of Rows
+
+	// pgConn is required in the connection to check connection status - we'll just set it to closed
+	pgConn := &pgconn.PgConn{}
+	pointerValPgConn := reflect.ValueOf(pgConn)
+	valPgConn := reflect.Indirect(pointerValPgConn)
+
+	fieldStatus := valPgConn.FieldByName("status")
+	ptrToStatus := unsafe.Pointer(fieldStatus.UnsafeAddr())
+	realPtrToStatus := (*byte)(ptrToStatus)
+	// connStatusClosed == 2
+	*realPtrToStatus = byte(2)
+
+	c := &pgx.Conn{}
+	pointerValConn := reflect.ValueOf(c)
+	valConn := reflect.Indirect(pointerValConn)
+
+	// connection checks some config values, so it should be set
+	fieldConfig := valConn.FieldByName("config")
+	ptrToConfig := unsafe.Pointer(fieldConfig.UnsafeAddr())
+	realPtrToConfig := (**pgx.ConnConfig)(ptrToConfig)
+	*realPtrToConfig = &pgx.ConnConfig{}
+
+	fieldPgConn := valConn.FieldByName("pgConn")
+	ptrToPgConn := unsafe.Pointer(fieldPgConn.UnsafeAddr())
+	realPtrToPgConn := (**pgconn.PgConn)(ptrToPgConn)
+	*realPtrToPgConn = pgConn
+
+	// we'll actually get an error as the connection is closed but we do not care - all we need is instance of rows
```

Comment:
I’m not sure if you want to add documentation that this would break if they ever start returning `nil, err`, instead of a zero-value. It looks like they have no reason to start returning an invalid value… but you never know?